src/filesystem.cpp

```diff
@@ -705,54 +698,80 @@ void set_user_data_dir(std::string newprefdir)
 			ERR_FS << "Could not determine path to user's Documents folder! (" << std::hex << "0x" << res << std::dec << ") "
 				   << "User config/data directories may be unavailable for "
 				   << "this session. Please report this as a bug.";
-			user_data_dir = bfs::path(get_cwd()) / newprefdir;
+			dir = bfs::path(get_cwd()) / newprefdir;
 		} else {
 			bfs::path games_path = bfs::path(docs_path) / "My Games";
 			create_directory_if_missing(games_path);
 
-			user_data_dir = games_path / newprefdir;
+			dir = games_path / newprefdir;
 		}
 
 		CoTaskMemFree(docs_path);
 	}
 
-#else /*_WIN32*/
+	return dir;
+}
+#endif
 
-	std::string backupprefdir = ".wesnoth" + get_version_path_suffix();
+#ifdef __APPLE__
+bfs::path apple_userdata(const std::string& newprefdir, bool relative_ok)
+{
+	bfs::path dir;
+	std::string temp = newprefdir;
 
-#ifdef WESNOTH_BOOST_OS_IOS
-	char *sdl_pref_path = SDL_GetPrefPath("wesnoth.org", "iWesnoth");
-	if(sdl_pref_path) {
-		backupprefdir = std::string(sdl_pref_path) + backupprefdir;
-		SDL_free(sdl_pref_path);
+	if(temp.empty()) {
+		temp = ".wesnoth" + get_version_path_suffix();
+		relative_ok = true;
 	}
+
+	const char* home_str = getenv("HOME");
+	bfs::path home = home_str ? home_str : ".";
+
+	if(temp[0] == '/') {
+		dir = temp;
+	} else {
+		if(!relative_ok) {
+			// TRANSLATORS: translate the part inside <...> only
+			deprecated_message(_("--userdata-dir=<relative path>"),
+				DEP_LEVEL::FOR_REMOVAL,
+				{1, 17, 0},
+				_("Use absolute paths. Relative paths are deprecated because they are interpreted relative to $HOME"));
```

Comment:
I get that this isn't your fault (or at least not something that you changed in this PR), but seeing this makes me wonder why it's not more like: 

```c++
"--userdata-dir=<" + _("something^relative path") + ">"
```

Reply:
Also, why is this message only in the Apple code? Is that intended? Seems weird.

Reply:
What? It's in all of them.

Reply:
Huh? I didn't see it in the others though… but if it's in all of them then that also sounds like unfortunate code duplication.

Reply:
I'm not sure where you're looking then, but they are all there.